design/proposals/ksonnet-lib-simple-constructors.md

```diff
@@ -0,0 +1,109 @@
+# Deprecate ksonnet lib custom constructors
+
+Status: Pending
+
+Version: Alpha
+
+## Abstract
+
+We want to deprecate the custom resource constructors in ksonnet-lib. Instead of the custom constructors, developers will be encouraged to use the builder pattern to construct their objects. Constructors can be reintroduced in hand rolled libraries upstream.
+
+## Motivation
+
+* ksonnet-lib includes constructors for popular resources. In a few cases, these constructors are incomplete or include items which may not be required. The incompleteness and inconsistencies are a barrier to adoption because developers do not know how to get started in a productive way.
+* ksonnet-lib should be a direct translation of the OpenAPI swagger.json.
+
+## Goal
+
+* Developers should be able to construct objects using the builder pattern rather than constructors.
+* Upstream projects should have a clean slate to build abstraction upon.
+
+## Non Goals
+
+The following are related but not the goals of this specific proposal
+
+* Create upstream projects that implement constructors (e.g. v1 Deployment)
+
+## Proposal
+
+The change would be introducing a new constructor for all resources.
+
+```jsonnet
+group:: {
+  v1:: {
+    local apiVersion = {apiVersion: "group/v1"},
+    kind:: {
+      local kind = {kind: "Kind"},
+      init() :: apiVersion + kind,
+    },
+  },
+}
+```
+
+Many of the existing constructors adhere to this format. This change would make sure all resources have a constructor that only includes the apiVersion and kind.
```

Comment:
nit: if it's not too much trouble, backticks around `apiVersion` and `kind` would be helpful